lib/x509Support.py

```diff
@@ -8,7 +8,7 @@
 from . import defaults
 
 
-def extract_DN(fname):
+def extract_DN(fname, exit_if_bad=True):
```

Comment:
This library is called in many places. 
If the file is missing or there is a different content probably it should not be called instead of ignoring the error.
I think the problem was to have an sys.exit inside a library function.
I'd prefer to:
1. leave `def extract_DN(fname):`
2. change original `sys.exit(3)` to a `raise` (re-raising ValueError)
3. catch ValueError in the caller
Would be nice to add a message also to the log file, but I'm not sure the logger is available